Services/Common/OJS.Workers/OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs

```diff
@@ -325,13 +320,34 @@ private static string EvaluateMavenTestOutput(string testOutput, Regex testError
             return GetMavenErrorsComment(testOutput);
         }
 
+        private void ValidateSubmissionFile(string submissionFilePath)
+        {
+            // Main class is validated against the value in pom.xml.
+            // If package names don't correspond,
+            // then the tests are saved in a different folder structure (see how this.PackageName is exported)
+            // and when ran, tests are unable to find their correct configuration.
+            if (!this.ValidateMainClassFileName(submissionFilePath))
+            {
+                throw new ArgumentException($"Submission does not contain start class at: {this.GetMainClassFilePath()}. Check your folder structure and pom.xml start-class definition.");
+            }
+
+            if (!ValidateFolderStructure(submissionFilePath))
+            {
+                throw new ArgumentException($"Folder structure is invalid! The zip folder structure must contain files with path: {MainCodeFolderPattern} and pom.xml start-class definition.");
```

Comment:
Where are these exceptions cacherd, as I see you moved it below the `catch` clause?

Reply:
Validations used to happen below the invocation of `CreateSubmissionFile` here:

![image](https://github.com/user-attachments/assets/263eb878-924c-46c2-bb2f-0c279b7d905d)

I've moved them inside the CreateSubmissionFile > PrepareSubmissionFile in order to do the validation before other unesessary operations against the submission are done.